var/spack/repos/builtin/packages/bzip2/package.py

```diff
@@ -24,11 +25,16 @@ class Bzip2(Package, SourcewarePackage):
     version('1.0.7', sha256='e768a87c5b1a79511499beb41500bcc4caf203726fff46a6f5f9ad27fe08ab2b')
     version('1.0.6', sha256='a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd')
 
-    variant('shared', default=True, description='Enables the build of shared libraries.')
+    variant('shared', default=(sys.platform != 'win32'),
+            description='Enables the build of shared libraries.')
     variant('pic', default=False, description='Build static libraries with PIC')
     variant('debug', default=False, description='Enable debug symbols and disable optimization')
 
-    depends_on('diffutils', type='build')
+    # makefile.msc doesn't provide a shared recipe
+    conflicts('+shared', when='platform=windows',
+              msg='Windows makefile has no recipe for shared builds, use ~shared.')
+
+    # depends_on('diffutils', type='build')
```

Comment:
This will need to be added back conditionally: I assume you're doing things this way for now because there is no way to express `not` in the spec (e.g. you can't do `depends_on(diffutils, when='!windows')`). For now, you could enable it for the `darwin/cray/linux` platforms (since that accounts for all known platforms except Windows), or follow the approach you took in `hdf5`:

```
    if sys.platform != 'darwin' and sys.platform != 'win32':
         depends_on('numactl', when='+mpi+fortran')
```